certbot-ci/windows_installer_integration_tests/test_main.py

```diff
@@ -0,0 +1,60 @@
+import os
+import time
+import unittest
+import subprocess
+
+
+@unittest.skipIf(os.name != 'nt', reason='Windows installer tests must be run on Windows.')
+def test_it(request):
+    try:
+        subprocess.check_call(['certbot', '--version'])
+    except (subprocess.CalledProcessError, OSError):
+        pass
+    else:
+        raise AssertionError('Expect certbot to not be available in the PATH.')
+
+    try:
+        # Install certbot
+        subprocess.check_call([request.config.option.installer_path, '/S'])
+
+        # Assert certbot is installed and runnable
+        output = subprocess.check_output(['certbot', '--version'], universal_newlines=True)
+        assert 'certbot 1.' in output, 'Flag --version does not output a version.'
+
+        # Assert renew task is installed and ready
+        output = _ps('(Get-ScheduledTask -TaskName "Certbot Renew Task").State', capture_stdout=True)
+        assert output.strip() == 'Ready'
+
+        # Assert renew task is working
+        now = time.time()
+        _ps('Start-ScheduledTask -TaskName "Certbot Renew Task"')
+
+        status = 'Running'
+        while status != 'Ready':
+            status = _ps('(Get-ScheduledTask -TaskName "Certbot Renew Task").State', capture_stdout=True).strip()
+            time.sleep(1)
+
+        log_path = os.path.join('C:\\', 'Certbot', 'log', 'letsencrypt.log')
+
+        modification_time = os.path.getmtime(log_path)
+        assert now < modification_time, 'Certbot log file has not been modified by the renew task.'
+
+        with open(log_path) as file_h:
+            data = file_h.read()
+        assert 'DEBUG:certbot._internal.renewal:no renewal failures' in data, 'Renew task did not execute properly.'
+
+    finally:
+        # Sadly this command cannot work in non interactive mode: uninstaller will ask explicitly permission in an UAC prompt
```

Comment:
That's unfortunate. I assume you didn't find any (sane) way to uninstall the program non-interactively?

If not, maybe we could either allow this code to run interactively or make it harder for a dev to run these tests and unknowingly make lasting changes to their machine? This could be done in future work if you prefer, but I think we should at least make an issue about it.

Reply:
This test is mainly designed to be run non interactively on Azure Pipelines. I propose we check for specific env variables when run on Azure, and warn at the initialization of pytest we explicit approval before running the test.

Reply:
Done, this test now asks for user confirmation. `--confirm` flag is added to execute the test non interactively with an implicit confirmation, and is used in the Azure pipeline.